distsql/select_result.go

```diff
@@ -153,6 +154,8 @@ type selectResult struct {
 	memTracker       *memory.Tracker
 
 	stats *selectResultRuntimeStats
+	mu    sync.Mutex
+	exit  bool
```

Comment:
could we use sync/atomic here?

Reply:
The atomic seems not enough here. When there is a response, if we check the atomic variable and make sure it's closed yet, then we start updating the consuming memory and copr stats, just before the update is executing, the result is closed and there may be a nil dereference.